superset/assets/src/explore/controls.jsx

```diff
@@ -103,6 +103,10 @@ const groupByControl = {
   optionRenderer: c => <ColumnOption column={c} showType />,
   valueRenderer: c => <ColumnOption column={c} />,
   valueKey: 'column_name',
+  filterOption: (opt, text) => (
```

Comment:
why does this only apply to the group by control? shouldn't we do the same for metric control?

Reply:
`MetricsControl` has it's own control that internally defines `filterOption` in it's own way. I alter `MetricsControl.selectFilterOption` in this very PR to address this.